public/app/plugins/panel/geomap/style/markers.ts

```diff
@@ -0,0 +1,257 @@
+          points: 4,
+          radius,
+          radius2: 0,
+          angle: 0,
+        }),
+      });
+    },
+  },
+  {
+    id: RegularShapeId.x,
+    name: 'X',
+    aliasIds: [MarkerShapePath.x],
+    make: (cfg: StyleConfigValues) => {
+      const radius = cfg.size ?? DEFAULT_SIZE;
+      return new Style({
+        image: new RegularShape({
+          stroke: new Stroke({ color: cfg.color, width: cfg.lineWidth ?? 1 }),
+          points: 4,
+          radius,
+          radius2: 0,
+          angle: Math.PI / 4,
+        }),
+      });
+    },
+  },
+];
+
+async function prepareSVG(url: string): Promise<string> {
+  return fetch(url, { method: 'GET' })
+    .then((res) => {
+      return res.text();
+    })
+    .then((text) => {
+      const parser = new DOMParser();
+      const doc = parser.parseFromString(text, 'image/svg+xml');
+      const svg = doc.getElementsByTagName('svg')[0];
+      if (!svg) {
+        return '';
+      }
+      // open layers requires a white fill becaues it uses tint to set color
+      svg.setAttribute('fill', '#fff');
+      const svgString = new XMLSerializer().serializeToString(svg);
+      const svgURI = encodeURIComponent(svgString);
+      return `data:image/svg+xml,${svgURI}`;
+    })
+    .catch((error) => {
+      console.error(error);
+      return '';
+    });
+}
+
+// Really just a cache for the various symbol styles
+const markerMakers = new Registry<SymbolMaker>(() => makers);
+
+export function getMarkerAsPath(shape?: string): string | undefined {
+  const marker = markerMakers.getIfExists(shape);
+  if (marker?.aliasIds?.length) {
+    return marker.aliasIds[0];
+  }
+  return undefined;
+}
+
+// Will prepare symbols as necessary
+export async function getMarkerMaker(symbol?: string): Promise<StyleMaker> {
+  if (!symbol) {
+    return circleMarker;
+  }
+
+  let maker = markerMakers.getIfExists(symbol);
+  if (maker) {
+    return maker.make;
+  }
+
+  // Prepare svg as icon
+  if (symbol.endsWith('.svg')) {
+    const src = await prepareSVG(getPublicOrAbsoluteUrl(symbol));
+    maker = {
+      id: symbol,
+      name: symbol,
+      aliasIds: [],
+      make: src
+        ? (cfg: StyleConfigValues) => {
+            const radius = cfg.size ?? DEFAULT_SIZE;
+            return [
+              new Style({
+                image: new Icon({
+                  src,
+                  color: cfg.color,
+                  opacity: cfg.opacity ?? 1,
+                  scale: (DEFAULT_SIZE + radius) / 100,
+                }),
+              }),
+              // transparent bounding box for featureAtPixel detection
+              new Style({
+                image: new RegularShape({
+                  fill: new Fill({ color: 'rgba(0,0,0,0)' }),
+                  points: 4,
+                  radius: cfg.size,
+                  angle: Math.PI / 4,
+                }),
+              }),
+            ];
+          }
+        : errorMarker,
+    };
+    markerMakers.register(maker);
```

Comment:
each URL will only try to load once :taco:

Reply:
Were URLs loading every time we were rendering before?